gulpfile.js

```diff
@@ -63,128 +63,134 @@ var unminifiedRuntimeTarget = 'dist/amp.js';
+declareExtension('amp-date-picker', '0.1', {hasCss: true});
+declareExtension('amp-image-viewer', '0.1', {hasCss: true});
 
 /**
  * @deprecated `amp-slides` is deprecated and will be deleted before 1.0.
  * Please see {@link AmpCarousel} with `type=slides` attribute instead.
  */
-declareExtension('amp-slides', '0.1', false);
-declareExtension('amp-social-share', '0.1', true);
-declareExtension('amp-timeago', '0.1', false);
-declareExtension('amp-twitter', '0.1', false);
-declareExtension('amp-user-notification', '0.1', true);
-declareExtension('amp-vimeo', '0.1', false);
-declareExtension('amp-vine', '0.1', false);
-declareExtension('amp-viz-vega', '0.1', true);
-declareExtension('amp-google-vrview-image', '0.1', false);
+declareExtension('amp-slides', '0.1', {hasCss: false});
+declareExtension('amp-social-share', '0.1', {hasCss: true});
+declareExtension('amp-timeago', '0.1', {hasCss: false});
+declareExtension('amp-twitter', '0.1', {hasCss: false});
+declareExtension('amp-user-notification', '0.1', {hasCss: true});
+declareExtension('amp-vimeo', '0.1', {hasCss: false});
+declareExtension('amp-vine', '0.1', {hasCss: false});
+declareExtension('amp-viz-vega', '0.1', {hasCss: true});
+declareExtension('amp-google-vrview-image', '0.1', {hasCss: false});
 declareExtension('amp-viewer-integration', '0.1', {
   // The viewer integration code needs to run asap, so that viewers
   // can influence document state asap. Otherwise the document may take
   // a long time to learn that it should start process other extensions
   // faster.
   loadPriority: 'high',
 });
-declareExtension('amp-video', '0.1', false);
-declareExtension('amp-vk', '0.1', false);
-declareExtension('amp-youtube', '0.1', false);
+declareExtension('amp-video', '0.1', {hasCss: false});
+declareExtension('amp-vk', '0.1', {hasCss: false});
+declareExtension('amp-youtube', '0.1', {hasCss: false});
 declareExtensionVersionAlias(
     'amp-sticky-ad', '0.1', /* lastestVersion */ '1.0', /* hasCss */ true);
+
+
+/**
+ * @typedef {{
+ *   name: ?string,
+ *   version: ?string,
+ *   hasCss: ?boolean,
+ *   loadPriority: ?string
+ * }}
+ */
+const ExtensionOption;
+
 /**
  * @param {string} name
  * @param {string} version E.g. 0.1
- * @param {boolean|!Object} hasCssOrOptions Whether the extension comes with CSS
- *   or an extension options object.
+ * @param {!ExtensionOption} options extension options object.
  */
-function declareExtension(name, version, hasCssOrOptions) {
-  var hasCss = false;
-  var options = {};
-  if (typeof hasCssOrOptions == 'boolean') {
-    hasCss = hasCssOrOptions;
-  } else {
-    options = hasCssOrOptions
-  }
-  extensions[name + '-' + version] = Object.assign({
+function declareExtension(name, version, options) {
+  const defaultOptions = {hasCss: false};
+  const extensionKey = `${name}-${version}`;
+  extensions[extensionKey] = Object.assign({
     name: name,
     version: version,
     hasCss: hasCss,
```

Comment:
Can remove this line, hasCss isn't available anymore in this scope!

Reply:
thanks!